lib/linguist/heuristics.yml

```
@@ -840,3 +846,20 @@ named_patterns:
   raku: '^\s*(?:use\s+v6\b|\bmodule\b|\b(?:my\s+)?class\b)'
   vb-class: '^\s*VERSION\s+[0-9]\.[0-9]\s+CLASS'
   vb-form: '^\s*VERSION\s+[0-9]\.[0-9]{2}'
+  vb-module: '^\s*Attribute\s+VB_Name\s+=\s+'
+  vba:
+  # VBA7 new 64-bit features (https://learn.microsoft.com/en-us/office/vba/language/concepts/getting-started/64-bit-visual-basic-for-applications-overview)
+  - '^\s*(?:Public|Private)?\s+Declare\s+PtrSafe\s+(?:Sub|Function)\b'
+  - '^\s*#If\s(:?VBA7|Win64)\b'
+  - '^\s*Dim\s[0-9a-zA-Z_]*\sAs\s(?:LongPtr|LongLong)\b'
+  # Top module declarations unique to VBA
+  - '^\s*Option\sPrivate\sModule\b'
+  - '^\s*Option\sCompare\sDatabase\b'
+  # Libraries/Objects preloaded with VBA
+  - '(?:\s|\()(?:VBA|Access|Excel|Outlook|PowerPoint|Visio|Word)\.\w+' # VBA Libraries invocation
+  - '\b(?:ThisDrawing|AcadObject)\b' # AutoCAD Objects
+  - '\b(?:ThisWorkbook|ActiveWorkbook|ActiveSheet|ActiveChart|ActiveCell|WorksheetFunction)\b' # Excel Objects
+  - '\bRange\("(?:[a-zA-Z]+\d+|[a-zA-Z]+\d+:[a-zA-Z]+\d+|[a-zA-Z]+:[a-zA-Z]+|\d+:\d+)"\)' # Excel range
+  - '\b(?:ActiveExplorer|ActiveInspector)\b' # Outlook Objects
+  - '\b(?:ActiveWindow\.Presentation|ActivePresentation)\b' # PowerPoint Objects
+  - '\b(?:ActiveDocument|Selection\.Find|Selection\.Paragraphs)\b' # Word Objects
```

We can combine most of these patterns, including their comments:

```suggestion
  # Objects used in AutoCAD, Outlook, PowerPoint, and Word
  - '\b(?:AcadObject|Active(?:Cell|Chart|Document|Explorer|Inspector|Presentation|Sheet|Window\.Presentation|Workbook)|Selection\.(?:Find|Paragraphs)|This(?:Drawing|Workbook)|WorksheetFunction)\b'
  # Excel range
  - '\bRange\("(?:[A-Za-z]+\d+(?::[A-Za-z]+\d+)?|[A-Za-z]+:[A-Za-z]+|[A-Za-z]*\d+(:\d+)?)"\)'
```

---

I'm thinking this section could be contentious and subject to future fine-tuning, so I'd keep them seperate for now just to keep things easier to edit.

---

Fair enough. In that case, I recommend moving the comments (`# … Objects` and friends) to their own lines, above the lines they're documenting, so they won't be missed by users reading the file in a split-pane editor window (or with soft-wrapping disabled).

---

Ok, I've moved the comments and I've also merged Outlook, PowerPoint and Word objects, but kept CAD objects seperate for now as it seems justified to keep it seperate from other Office apps.